### PR TITLE
348947 outputs from reprocessing

### PR DESCRIPTION
**Pull request by AtosSaqlain**

### CODE REVIEW

File: src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Controllers/OverseasSiteController.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    using Microsoft.AspNetCore.Mvc;
+
+    [Route("Accreditation/{accreditationExternalId}/[controller]/")]
+    public class OverseasSiteController : Controller
+    {
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly IAccreditationService _accreditationService;
+        private readonly ISaveAndComeBackService _saveAndComeBackService;
+        private readonly BackPageViewModel _backPageViewModel;
+        private IUrlHelperWrapper _urlHelper;
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Controllers/OverseasSiteController.cs]: This should also be readonly

File: src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Controllers/OverseasSiteController.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            if (accreditationExternalId.HasValue && siteExternalId.HasValue)
+            {
+                var overseasSiteOutputsViewModel = await _accreditationService.GetOverseasReprocessingSiteOutputs(
+                    accreditationExternalId.Value,
+                    siteExternalId.Value);
+
+                return View(overseasSiteOutputsViewModel);
+            }
+
+            return BadRequest("Missing over seas site ids");
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Controllers/OverseasSiteController.cs]: We've been returning NotFound if the ids are not present.

File: src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Controllers/OverseasSiteController.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            if (saveButton == SaveButton.SaveAndContinue && string.IsNullOrWhiteSpace(overseasReprocessingSiteOutputsViewModel.Outputs))
+            {
+                return RedirectToAction("OverseasSiteOutput", "OverseasSite");
+            }
+            else if (saveButton == SaveButton.SaveAndContinue && !string.IsNullOrWhiteSpace(overseasReprocessingSiteOutputsViewModel.Outputs))
+            {
+                return RedirectToAction("RejectedWastePlans", "Accreditation");
+            }
+
+            await _saveAndComeBackService.AddSaveAndComeBack(
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Controllers/OverseasSiteController.cs:161]: This only appears to do SaveAndComeBack. Where's the logic for moving to the next page?

File: src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Services/Accreditation/AccreditationService.cs [hunk 4]
```diff
@@ -87,6 +88,24 @@ public async Task<CheckYourAnswersViewModel> CheckYourAnswers(Guid id)
...
 
+        public async Task<OverseasReprocessingSiteOutputsViewModel> GetOverseasReprocessingSiteOutputs(
+            Guid accreditationExternalId,
+            Guid overseasSiteExternalId)
+        {
+            var overseasSiteOutputs = await _httpAccreditationService.GetOverseasReprocessingSiteOutputs(
+                accreditationExternalId,
+                overseasSiteExternalId);
+            var vm = _mapper.Map<OverseasReprocessingSiteOutputsViewModel>(overseasSiteOutputs);
+            vm.Id = accreditationExternalId;
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Services/Accreditation/AccreditationService.cs]: Is this needed to be provided on the Get? I've always put the Id property on the view model for doing the updates, and that gets automatically populated from the route

File: src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Controllers/OverseasSiteController.cs (hunk 3 above)
Comment (AtosSaqlain) [reply src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Controllers/OverseasSiteController.cs:161]: Isn't it on the lines above?

Comment (NikPeters-Atos) [reply src/EPR.Accreditation.Portal/EPR.Accreditation.Portal/Controllers/OverseasSiteController.cs:161]: Apologies!

**Review by NikPeters-Atos — APPROVED** [review]

**Pull request merged**